### ISSUE TITLE
[Question] Wandb metrics are zero for all but every 16th step

### ISSUE THREAD

**Issue by commotum** [sapientinc/HRM#48]
I'm currently working on reproducing your results and have a question about the expected logging behavior during training.

Describe the issue
I'm observing a pattern in my wandb logs where most metrics (train/accuracy, train/steps, train/count, etc.) are zeroed out for 15 consecutive steps and then report a value only on every 16th step. This is happening consistently throughout my training run.

Here's a look at my wandb graphs, which clearly show this sawtooth pattern:

<img width="1128" height="914" alt="Image" src="https://github.com/user-attachments/assets/dbcac53c-7c8a-4b64-bd1c-b948d6fc67d2" />

Here's the args I'm passing:
OMP_NUM_THREADS=8 python pretrain.py \
  data_path=data/sudoku-extreme-25k-aug-1000 \
  epochs=20000 \
  eval_interval=2000 \
  global_batch_size=2048 \
  lr=7e-5 \
  puzzle_emb_lr=7e-5 \
  weight_decay=0.01 \
  puzzle_emb_weight_decay=0.01

Question

Is this the intended logging behavior? I expected to see metrics updated at each step within a segment, rather than only at the end.

If this is not the intended behavior, could you point me toward what I might have misconfigured? I'm trying to understand if the model is learning correctly at each step or if something is preventing updates until the end of a segment.

I've attached a snippet of my CSV logs for train/accuracy and train/steps to provide a more concrete example of what I'm seeing.

Thank you for any help you can offer.

[wandb_export_2025-08-08T23_31_54.556-07_00.csv](https://github.com/user-attachments/files/21696415/wandb_export_2025-08-08T23_31_54.556-07_00.csv)
[wandb_export_2025-08-08T23_32_20.285-07_00.csv](https://github.com/user-attachments/files/21696414/wandb_export_2025-08-08T23_32_20.285-07_00.csv)
[wandb_export_2025-08-08T23_32_52.738-07_00.csv](https://github.com/user-attachments/files/21696413/wandb_export_2025-08-08T23_32_52.738-07_00.csv)
[wandb_export_2025-08-08T23_33_02.983-07_00.csv](https://github.com/user-attachments/files/21696416/wandb_export_2025-08-08T23_33_02.983-07_00.csv)
